### PR TITLE
Normalize error handling

### PR DESCRIPTION
**Pull request by bcrochet**
Fix the error handling in three packages:
* artifacts
* external engine
* formatters
    
Just going alphabetical through the packages and normalizing error
handling, and removing errors.go from packages when and where
appropriate.
    
Signed-off-by: Brad P. Crochet <brad@redhat.com>

### CODE REVIEW

**Comment by bcrochet** [comment]
Hold until #652 lands

File: certification/artifacts/artifacts.go [hunk 1]
```diff
@@ -17,7 +18,7 @@ func WriteFile(filename, contents string) (string, error) {
 
 	err := os.WriteFile(fullFilePath, []byte(contents), 0o644)
 	if err != nil {
-		return fullFilePath, err
+		return fullFilePath, fmt.Errorf("could not write file to artifacts diretory: %v", err)
```
Comment (acornett21) [inline certification/artifacts/artifacts.go:21]: Should this be `%w` for consistency? `%v and %w` are synonyms, but maybe we should pick one and stick with it?

File: certification/formatters/generic.go [hunk 2]
```diff
@@ -39,8 +37,7 @@ func genericXMLFormatter(ctx context.Context, r runtime.Results) ([]byte, error)
 
 	responseJSON, err := xmlMarshalIndent(response, "", "    ")
 	if err != nil {
-		e := fmt.Errorf("%w with formatter %s: %s",
-			errors.ErrFormattingResults,
+		e := fmt.Errorf("error formatting results with formatter %s: %w",
 			"json",
```
Comment (acornett21) [inline certification/formatters/generic.go:41]: Should this be `xml`?

Comment (komish) [reply certification/formatters/generic.go:41]: also `responseXML?`

File: certification/internal/engine/engine.go [hunk 3]
```diff
@@ -327,27 +326,27 @@ func untar(dst string, r io.Reader) error {
...
 	manifest, err := imageRef.ImageInfo.Manifest()
 	if err != nil {
-		return fmt.Errorf("%w: %s", errors.ErrImageInspectFailed, err)
+		return fmt.Errorf("%w: %s", ErrImageInspectFailed, err)
 	}
 
 	digest, err := imageRef.ImageInfo.Digest()
 	if err != nil {
-		return fmt.Errorf("%w: %s", errors.ErrImageInspectFailed, err)
+		return fmt.Errorf("%w: %s", ErrImageInspectFailed, err)
```
Comment (komish) [inline certification/internal/engine/engine.go:339]: Does this need to be `("%s: %w", ...)`? Or was that done in another PR?

**Review by komish — COMMENTED** [review]
Small things.

File: certification/artifacts/artifacts.go (hunk 1 above)
Comment (acornett21) [reply certification/artifacts/artifacts.go:21]: `%v` means `default` so I assume in the [docs](https://pkg.go.dev/fmt#Errorf) just mean `%w` is the same as `default`, at least that's how i read it.

Comment (bcrochet) [reply certification/artifacts/artifacts.go:21]: They aren't exactly synonyms. %w allows for unwrapping the error, whereas %v does not. Not sure what you mean by default.

File: certification/formatters/generic.go (hunk 2 above)
Comment (bcrochet) [reply certification/formatters/generic.go:41]: That looks like that's been there a while. I'll fix though.

File: certification/artifacts/artifacts.go (hunk 1 above)
Comment (bcrochet) [reply certification/artifacts/artifacts.go:21]: And %v was chosen precisely so that the error type doesn't leak out of the package. If %w is returned from a package export, it will end up being a part of the API, and we do not want to do that unless we are explicit about it.

File: certification/formatters/generic.go (hunk 2 above)
Comment (acornett21) [reply certification/formatters/generic.go]: Thanks for cleaning this up too!

**Review by acornett21 — APPROVED** [review]
